l3kernel/l3basics.dtx

```diff
@@ -3382,45 +3389,121 @@
-\cs_new:Npn \cs_prefix_spec:N #1
+
+\cs_if_exist:NTF \@@_macro_prefix_spec:N
   {
-    \token_if_macro:NTF #1
+    \cs_new:Npn \cs_prefix_spec:N #1
       {
-        \exp_after:wN \@@_prefix_arg_replacement:wN
-          \token_to_meaning:N #1 \s_@@_stop \use_i:nnn
+        \token_if_macro:NTF #1
+          { \@@_macro_prefix_spec:N #1 }
+          { \scan_stop: }
       }
-      { \scan_stop: }
   }
-\cs_new:Npn \cs_parameter_spec:N #1
   {
-    \token_if_macro:NTF #1
+    \cs_new:Npn \cs_prefix_spec:N #1
       {
-        \exp_after:wN \@@_prefix_arg_replacement:wN
-          \token_to_meaning:N #1 \s_@@_stop \use_ii:nnn
+        \token_if_macro:NTF #1
+          {
+            \exp_after:wN \@@_prefix_arg_replacement:wN
+              \token_to_meaning:N #1 \s_@@_stop \use_i:nnn
+          }
+          { \scan_stop: }
       }
-      { \scan_stop: }
   }
-\cs_new:Npn \cs_replacement_spec:N #1
+  \cs_if_exist:NTF \@@_macro_parameter_spec:N
   {
-    \token_if_macro:NTF #1
+    \cs_new:Npn \cs_parameter_spec:N #1
       {
-        \exp_after:wN \@@_prefix_arg_replacement:wN
-          \token_to_meaning:N #1 \s_@@_stop \use_iii:nnn
+        \token_if_macro:NTF #1
+        { \@@_macro_parameter_spec:N #1 }
+          { \scan_stop: }
       }
-      { \scan_stop: }
   }
+  {
+    \cs_new:Npn \cs_parameter_spec:N #1
+      {
+        \token_if_macro:NTF #1
+          {
+            \exp_after:wN \@@_prefix_arg_replacement:wN
+              \token_to_meaning:N #1 \s_@@_stop \use_ii:nnn
+          }
+          { \scan_stop: }
+      }
+  }
+\cs_if_exist:NTF \@@_macro_replacement_spec:N
+  {
+    \cs_new:Npn \cs_replacement_spec:N #1
+      {
+        \token_if_macro:NTF #1
+          { \@@_macro_replacement_spec:N #1 }
+          { \scan_stop: }
+      }
+  }
+  {
+    \cs_new:Npn \cs_replacement_spec:N #1
+      {
+        \token_if_macro:NTF #1
+          {
+            \exp_after:wN \@@_prefix_arg_replacement:wN
+              \token_to_meaning:N #1 \s_@@_stop \use_iii:nnn
+          }
+          { \scan_stop: }
+      }
+  }
+%</code>
+%    \end{macrocode}
+%   This is a modified version of code suggested by Max Chernoff
+%   (\url{https://chat.stackexchange.com/transcript/message/67947633#67947633}).
+%   We aim to provide the classical \TeX{}82 appearance but covering \eTeX{}:
+%   due to changes in the engine, the only useful prefix here is |\protected|
+%   at present. (This may need to be revisited once the engine is more stable.)
+%   This does not cover tokens other than control sequences, so we only use
+%   it for the internal needed here.
+%    \begin{macrocode}
+%<*lua>
+if status.luatex_engine == 'luametatex' then
+  local function scan_full_csname()
+    local t = get_next()
+    local csname = get_csname(t)
+    return t.active and active_prefix .. csname or csname, t
+  end
+  luacmd('@@_macro_prefix_spec:N', function()
+    local token = get_next()
+    if get_protected(token) then
+      sprint(-2, "\\protected ")
```

Comment:
The other strings in this PR use `'single quotes'`, so maybe this one should too?

```suggestion
      sprint(-2, '\\protected ')
```

Reply:
Ah, that makes sense then (and seems reasonable to me too).